app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesFragment.java

```diff
@@ -35,61 +47,96 @@
+     * this fragment using the provided parameters.
+     *
+     * @param eventId Parameter 1.
+     * @return A new instance of fragment AttendeesFragment.
+     */
+    // TODO: Rename and change types and number of parameters
+    public static AttendeesFragment newInstance(long eventId) {
+        AttendeesFragment fragment = new AttendeesFragment();
+        Bundle args = new Bundle();
+        args.putLong(EventsActivity.EVENT_KEY, eventId);
+        fragment.setArguments(args);
+        return fragment;
+    }
 
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
+    public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_attendee_list);
-
-        ButterKnife.bind(this);
+        if (getArguments() != null) {
+            eventId = getArguments().getLong(EventsActivity.EVENT_KEY);
+            attendeesPresenter = new AttendeesPresenter(eventId, this, new EventDataRepository(new UtilModel(activity)));
+        }
+    }
 
-        Intent i = getIntent();
-        long id = i.getLongExtra("id", 0);
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        View view = inflater.inflate(R.layout.fragment_attendees, container, false);
+        ButterKnife.bind(this, view);
+        return view;
+    }
 
-        attendeesPresenter = new AttendeesPresenter(id, this, new EventDataRepository(new UtilModel(this)));
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
 
-        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(this);
-        attendeeListAdapter = new AttendeeListAdapter(this, attendeesPresenter);
+        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(activity);
+        attendeeListAdapter = new AttendeeListAdapter(activity, attendeesPresenter);
         recyclerView.setAdapter(attendeeListAdapter);
         recyclerView.setLayoutManager(layoutManager);
 
-        attendeesPresenter.attach();
-
         btnBarCodeScanner.setOnClickListener(v -> {
-            Intent scanQr = new Intent(AttendeesActivity.this, ScanQRActivity.class);
-            scanQr.putExtra(ATTENDEES_KEY, id);
-            startActivityForResult(scanQr, 123);
+            Intent scanQr = new Intent(activity, ScanQRActivity.class);
+            scanQr.putExtra(EventsActivity.EVENT_KEY, eventId);
+            startActivityForResult(scanQr, REQ_CODE);
         });
+
+        attendeesPresenter.attach();
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        activity = context;
     }
 
     @Override
-    protected void onDestroy() {
-        super.onDestroy();
+    public void onDetach() {
```

Comment:
That was raising npe for `attendeesView` on switching between the fragments while async task is started. Since it would make `attendeesView` null in `onDetach`.